### PR TITLE
OCPBUGS-5834:specifies naming parameters for SR-IOV resourceName

### PR DESCRIPTION
**Pull request by JoeAldinger**
OCPBUGS-5834: specifies naming conventions for SR-IOV `resourceName` 

Version(s):
4.10+

Issue:
https://issues.redhat.com/browse/OCPBUGS-5834

Link to docs preview:
https://55447--docspreview.netlify.app/openshift-enterprise/latest/networking/hardware_networks/configuring-sriov-device.html

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/nw-sriov-networknodepolicy-object.adoc [hunk 1]
```diff
@@ -41,6 +41,8 @@ spec:
 <2> The namespace where the SR-IOV Network Operator is installed.
 
 <3> The resource name of the SR-IOV network device plugin. You can create multiple SR-IOV network node policies for a resource name.
++
+When specifying a name, be sure to not use special characters such as `-` in the `resourceName`.
```
Comment (zhaozhanqi) [inline modules/nw-sriov-networknodepolicy-object.adoc:45]: How about the correct resource name express is `^[a-zA-Z0-9_]+$`

Comment (zhaozhanqi) [reply modules/nw-sriov-networknodepolicy-object.adoc:45]: if the resource name contain '-', the webhook can blocked it and return error, eg

Error from server (resource name "e810-c" contains invalid characters, the accepted syntax of the regular expressions is: "^[a-zA-Z0-9_]+$"): error when creating "snnp.yaml": admission webhook "operator-webhook.sriovnetwork.openshift.io" denied the request: resource name "e810-c" contains invalid characters, the accepted syntax of the regular expressions is: "^[a-zA-Z0-9_]+$"
 
unless the customer disable the webhook feature.

Comment (JoeAldinger) [reply modules/nw-sriov-networknodepolicy-object.adoc]: @zhaozhanqi I've updated based on your feedback. Does the update to the document work?

**Pull request merged**